src/server/journal/journal_slice.cc

```diff
@@ -116,36 +117,63 @@ error_code JournalSlice::Close() {
 }
 #endif
 
+bool JournalSlice::IsLSNInBuffer(LSN lsn) const {
+  DCHECK(ring_buffer_);
+
+  if (ring_buffer_->empty()) {
+    return false;
+  }
+  return (*ring_buffer_)[0].lsn <= lsn && lsn <= ((*ring_buffer_)[ring_buffer_->size() - 1].lsn);
```

Comment:
(*ring_buffer_)[0].lsn is the oldest journal entry that we saved correct?
If so I think we can write:
if ((*ring_buffer_)[0].lsn <= lsn) {
  DCHECK((*ring_buffer_)[ring_buffer_->size() - 1].lsn);
  return true;
}
return false;

Reply:
The replica can request to perform a partial sync from the future and I would have to reject it graciously somehow.

Reply:
So this will be a bug if replica request to perform a partial sync from the future right?
I think we better have a check fail on this in debug mode and on release just print an error on dont allow partial sync in this case

Reply:
discussed, DCHECK will be added in the partial sync PR